lib/canonical_version.sh

```diff
@@ -1,4 +1,28 @@
-canonicalVersion() {
+#!/usr/bin/env bash
+
+fetch_erlang_versions() {
+  url="https://raw.githubusercontent.com/HashNuke/heroku-buildpack-elixir-otp-builds/master/otp-versions"
+  curl -s "$url" 
+}
+
+exact_erlang_version_available() {
+  version=$1
+  available_versions=$2
+  found=1
+  while read -r line; do
+    if [ "$line" = "$version" ]; then
+      found=0
+    fi
+  done <<< "$available_versions"
+  echo $found
+}
+
+check_erlang_version() {
   version=$1
-  echo $version
+  exists=$(exact_erlang_version_available "$version" "$(fetch_erlang_versions)")
```

Comment:
why is `fetch_erlang_versions` here, instead of just assigned directly to `$available_versions` in `exact_erlang_version_available`?

Reply:
It's so that `exact_erlang_version_available` can remain a pure function that is deterministic, without side effects, and more easily testable.